SubsoniciousKit/SubsoniciousKit/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  AuthenticationManager.swift
+//  SubsoniciousKit
+//
+//  Created by Bilal on 05/07/2020.
+//  Copyright © 2020 Bilal Benlarbi. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+public class AuthenticationManager: ObservableObject {
+
+    private let authenticationService: AuthenticationService
+    @Published public private(set) var result: Result<SubsonicResponse, Error>?
+    private var cancellables: Set<AnyCancellable> = []
+
+    public init(authenticationService: AuthenticationService) {
+        self.authenticationService = authenticationService
+    }
+
+    public func authenticate(with server: Server) {
+        authenticationService.fetchAuthentication(with: server)
+            .map { Result<SubsonicResponse, Error>.success($0) }
```

Comment:
you could use an extension : 
```
func asResult() -> AnyPublisher<Result<Self.Output, Self.Failure>, Never> {
        self.map { .success($0) }
            .catch { Just(.failure($0)) }
            .eraseToAnyPublisher()
    }
```

Reply:
Resolving that conversation as it's duplicated from https://github.com/bilalb/Subsonicious/pull/5#discussion_r454268257